flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPoolTest.java

```diff
@@ -543,6 +543,164 @@ public void testConsistentAvailability() throws Exception {
         }
     }
 
+    @Test
+    public void testOverdraftBuffer() throws IOException, InterruptedException {
```

Comment:
Why do you keep all scenarios in one test? It looks too big and difficult to read. Can you please, split this test into several tests with the descriptive names?